internal/webhook/notebook/mutating.go

```diff
@@ -0,0 +1,310 @@
+		return admission.Errored(http.StatusBadRequest, fmt.Errorf("failed to decode object: %w", err)), false, nil
+	}
+
+	// Get the connections annotation
+	objAnnotations := obj.GetAnnotations()
+	if objAnnotations == nil {
+		// No annotations - allow the request
+		return admission.Allowed("No annotations found - connections validation skipped"), false, nil
+	}
+
+	connectionsValue, found := objAnnotations[annotations.Connection]
+	if !found {
+		// No connections annotation - allow the request
+		return admission.Allowed("No connections annotation found - connections validation skipped"), false, nil
+	}
+
+	if strings.TrimSpace(connectionsValue) == "" {
+		// Empty connections annotation - allow the request
+		return admission.Allowed("Empty connections annotation - connections validation skipped"), false, nil
+	}
+
+	// Parse the connections annotation
+	connectionSecrets, err := parseConnectionsAnnotation(connectionsValue)
+	if err != nil {
+		log.Error(err, "failed to parse connections annotation", "connectionsValue", connectionsValue)
+		return admission.Denied(fmt.Sprintf("failed to parse connections annotation: %v", err)), false, nil
+	}
+
+	// Validate permissions for each connection secret
+	var permissionErrors []string
+	secretRefs := make([]corev1.SecretReference, 0)
+	for _, secretRef := range connectionSecrets {
+		log.V(1).Info("checking permission for secret", "secret", secretRef.Name, "namespace", secretRef.Namespace)
+		hasPermission, err := w.checkSecretPermission(ctx, req, secretRef)
+		if err != nil {
+			log.Error(err, "error checking permission for secret", "secret", secretRef.Name, "namespace", secretRef.Namespace)
+			return admission.Errored(http.StatusInternalServerError, fmt.Errorf("error checking permission for secret %s/%s: %w", secretRef.Namespace, secretRef.Name, err)), false, nil
+		}
+
+		if !hasPermission {
+			permissionErrors = append(permissionErrors, fmt.Sprintf("%s/%s", secretRef.Namespace, secretRef.Name))
+		}
+
+		secretRefs = append(secretRefs, secretRef)
+	}
+
+	if len(permissionErrors) > 0 {
+		return admission.Denied(fmt.Sprintf("user does not have permission to access the following connection secrets: %s", strings.Join(permissionErrors, ", "))), false, nil
+	}
+
+	return admission.Allowed("Connection permissions validated successfully"), true, secretRefs
+}
+
+// parseConnectionsAnnotation parses the connections annotation value into a list of secret references.
+// The annotation value should be a comma-separated list of fully qualified secret names (namespace/name).
+func parseConnectionsAnnotation(value string) ([]corev1.SecretReference, error) {
+	if strings.TrimSpace(value) == "" {
+		return nil, nil
+	}
+
+	parts := strings.Split(value, ",")
+	secretRefs := make([]corev1.SecretReference, 0)
+
+	for _, part := range parts {
+		part = strings.TrimSpace(part)
+		if part == "" {
+			continue
+		}
+
+		// Parse namespace/name format
+		secretParts := strings.Split(part, "/")
```

Comment:
are the secrets expected to be always expressed as `namespace/name`  or just having `name` would identify a secret in the notebook namespace ? (just asking)